File: SpendingTracker-Mobile/app/src/main/java/vn/edu/fpt/spendingtracker_mobile/dtos/MonthlyAmountsByYearDto.java
package vn.edu.fpt.spendingtracker_mobile.dtos;

import java.math.BigDecimal;
import java.util.Map;

public class MonthlyAmountsByYearDto {
    public int year;
    public Map<Integer, BigDecimal> monthlyIncomes;
    public Map<Integer, BigDecimal> monthlyExpenses;

    public MonthlyAmountsByYearDto() {
    }

    public MonthlyAmountsByYearDto(int year, Map<Integer, BigDecimal> monthlyIncomes, Map<Integer, BigDecimal> monthlyExpenses) {
        this.year = year;
        this.monthlyIncomes = monthlyIncomes;
        this.monthlyExpenses = monthlyExpenses;
    }
}

File: SpendingTracker-API/DTOs/Statistics/MonthlyAmountsByYearDto.cs
namespace SpendingTracker_API.DTOs.Statistics
{
    public class MonthlyAmountsByYearDto
    {
        public int Year { get; set; }
        public Dictionary<int, decimal> MonthlyIncomes { get; set; } = new Dictionary<int, decimal>();
        public Dictionary<int, decimal> MonthlyExpenses { get; set; } = new Dictionary<int, decimal>();
    }
}
